### ISSUE TITLE
need a interface to get the node url 

### ISSUE THREAD

**Issue by shmilyoo** [olivere/elastic#1534]
Please use the following questions as a guideline to help me answer
your issue/question without further inquiry. Thank you.

### Which version of Elastic are you using?

[ ✔️ ] elastic.v7 (for Elasticsearch 7.x)
[ ✔️ ] elastic.v6 (for Elasticsearch 6.x)

### Please describe the expected behavior

The official nodejs client for es provides the interface 
```
esClient.connectionPool.getConnection() 
```

to get the connection in live state. These connectionPool will be updated based on the result of each sniff, which basically ensures that the obtained connection is a healthy es node.
Do we have this feature too?

**Comment by olivere** [comment]
What is the use case of this feature? I pretty much try to hide as much as I can from the outside. If you end up opening all the internals, people start relying on it, and you move yourself in a position where you cannot change anything anymore because people rely on it. Probably a special case of [Hyrum's Law](https://www.hyrumslaw.com/)

**Comment by shmilyoo** [comment]
> What is the use case of this feature? I pretty much try to hide as much as I can from the outside. If you end up opening all the internals, people start relying on it, and you move yourself in a position where you cannot change anything anymore because people rely on it. Probably a special case of [Hyrum's Law](https://www.hyrumslaw.com/)

Your principle makes sense, but I need features that have my own usage scenarios.

I need to make an es agent, and this is a business-required feature that later needs to be universally applied to all es in production.
For incoming es requests, it will be passed through to the actual es server. But I need to know the address and other information of the currently live es nodes. This requires the sniff function that comes with the esclient to get this information.

In other words, in addition to the implementation of the pass-through proxy function, I will maintain an esClient, generate this client according to the user's sniff and other configurations. Then, in each pass-through proxy request, I will get a connection from the connectionPool of this client to get the url of an es node to send this pass-through proxy request.

**Comment by olivere** [comment]
So the basic premise is to have a [reverse proxy server](https://pkg.go.dev/net/http/httputil#ReverseProxy) to an ES cluster. If that's correct, I don't think opening up the current state of the library would be the best option. Instead, you could periodically run HTTP Nodes Info API (`http://127.0.0.1:9200/_nodes/http?pretty=true` as [described here briefly](https://github.com/olivere/elastic/wiki/Sniffing)) and retrieve the node URLs yourself. If you don't handle that yourself, you end up being bound to the strategy that this library chooses to pick the next node to send requests to.

I mean, I _could_ give you access to the next URL that elastic would send a request to. But that might change on the next request, so you need to do health checks and retries anyway.

**Comment by shmilyoo** [comment]
> So the basic premise is to have a [reverse proxy server](https://pkg.go.dev/net/http/httputil#ReverseProxy) to an ES cluster. If that's correct, I don't think opening up the current state of the library would be the best option. Instead, you could periodically run HTTP Nodes Info API (`http://127.0.0.1:9200/_nodes/http?pretty=true` as [described here briefly](https://github.com/olivere/elastic/wiki/Sniffing)) and retrieve the node URLs yourself. If you don't handle that yourself, you end up being bound to the strategy that this library chooses to pick the next node to send requests to.
> 
> I mean, I _could_ give you access to the next URL that elastic would send a request to. But that might change on the next request, so you need to do health checks and retries anyway.

I can understand what you mean by implementing a simple sniff request myself and parsing the returned results.
So I'll do that.
